mplc/multi_partner_learning.py

```diff
@@ -654,3 +655,153 @@ def init_multi_partner_learning_from_scenario(scenario, is_save_data=True):
     )
 
     return mpl
+
+
+class MplLabelFlip(MultiPartnerLearning):
+
+    def __init__(self, scenario, is_save_data=False, epsilon=0.01):
+        super().__init__(
+            scenario.partners_list,
+            scenario.epoch_count,
+            scenario.minibatch_count,
+            scenario.dataset,
+            scenario.multi_partner_learning_approach,
+            scenario.aggregation_weighting,
+            scenario.is_early_stopping,
+            is_save_data,
+            scenario.save_folder,
+            scenario.init_model_from,
+            scenario.use_saved_weights,
+        )
+
+        self.epsilon = epsilon
+        self.K = scenario.dataset.num_classes
+        self.history_theta = [[None for _ in self.partners_list] for _ in range(self.epoch_count)]
+        self.history_theta_ = [[None for _ in self.partners_list] for _ in range(self.epoch_count)]
+        self.theta = [self.init_flip_proba() for _ in self.partners_list]
+        self.theta_ = [None for _ in self.partners_list]
+        # self.labels_map = LabelEncoder().fit_transform([str(y) for y in scenario.dataset.y_train])
```

Comment:
Is this going to be used some day? Just asking

Reply:
Egh not really, it was an attempt to make it agnostic to label type. But this problem must be adressed with deeper changes in the code.